pkg/manifestcomparators/comp_no_field_removal.go

```diff
@@ -22,6 +22,18 @@ func (noFieldRemoval) WhyItMatters() string {
 	return "If fields are removed, then clients that rely on those fields will not be able to read them or write them."
 }
 
+
+func getFields(version *apiextensionsv1.CustomResourceDefinitionVersion) sets.String {
+	fields := sets.NewString()
+	SchemaHas(version.Schema.OpenAPIV3Schema, field.NewPath("^"), field.NewPath("^"), nil,
+		func(s *apiextensionsv1.JSONSchemaProps, fldPath, simpleLocation *field.Path, _ []*apiextensionsv1.JSONSchemaProps) bool {
+			fields.Insert(simpleLocation.String())
+			return false
+		})
+
+	return fields
+}
+
```

Comment:
Not needed for this PR, but it seems like a good refactor.

Reply:
Agreed